django/contrib/postgres/fields/array.py

```diff
@@ -179,6 +180,14 @@ def as_sql(self, qn, connection):
 
 
 @ArrayField.register_lookup
+class ArrayExact(Exact):
+    def as_sql(self, qn, connection):
+        sql, params = super(ArrayExact, self).as_sql(qn, connection)
+        sql += '::%s' % self.lhs.output_field.db_type(connection)
```

Comment:
Considering this change isn't targeted specifically at CharField and should work for a variety of fields, it might be good to have some number/integer based tests as well.